src/index.js

```diff
@@ -0,0 +1,8 @@
+'use strict'
+
+module.exports = {
+  Stop: require('./circuit/hop'),
+  Dialer: require('./dialer'),
+  multicodec: require('./multicodec'),
+  tag: 'Circuit'
+}
```

Comment:
Yeah, there really isn't currently any other way of achieving that without riding the code with ad-hoc changes all over the place. 

Also after actually working with the transport interface it ended up covering all of circuit's use cases, because at the end and once you introduce an address plus dialing/listening logic you do inherit most of other transport's traits. I'm still not 100% sold on doing it this way, but after thinking what this new pipeline gonna look like and then laying that on top of circuits requirements I keep coming back to the transport interface as something that covers all those use cases and does it In a consistent way. 

We should definitely expose a lower level interface that allows hooking into the connect/dial flow for other uses.